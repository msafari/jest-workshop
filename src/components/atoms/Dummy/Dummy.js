import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import 'jest-styled-components'

const propTypes = {
  dummyProp: PropTypes.number.isRequired,
  className: PropTypes.string
}

const StyledDummy = styled.div`
  color: red;
  width: 100%;
`

const Dummy = ({
  className,
  dummyProp
}) => {
  return (
    <StyledDummy
      className={className}
    >
      My fake dummy prop is {dummyProp}
    </StyledDummy>
  )
}

Dummy.propTypes = propTypes

export default Dummy
