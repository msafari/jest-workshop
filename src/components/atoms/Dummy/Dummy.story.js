import React from 'react'
import { storiesOf } from '@storybook/react'
import Dummy from '.'

const story = storiesOf('atoms/dummy', module)
  .add('without props', () => {
    return (<Dummy/>)
  })
  .add('with required props', () => {
    return (<Dummy dummyProp={40}/>)
  })

export default story
