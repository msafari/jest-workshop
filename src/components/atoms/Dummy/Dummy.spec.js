import React from 'react'
import render from 'react-test-renderer'
import Dummy from '.'

let dummyObj, dummy

describe('Dummy Component', () => {
  test('renders correctly with no props', () => {
    dummyObj = (<Dummy />)
    dummy = render.create(dummyObj).toJSON()
    expect(console.error).toHaveBeenCalled()
  })

  test('renders correctly with required props', () => {
    dummyObj = (<Dummy dummyProp={40}/>)
    dummy = render.create(dummyObj).toJSON()
    expect(dummy).toMatchSnapshot()
    expect(dummy).toHaveStyleRule('color', 'red')
    expect(dummy).toHaveStyleRule('width', '100%')
  })
})
