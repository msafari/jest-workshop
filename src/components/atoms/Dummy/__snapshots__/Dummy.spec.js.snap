// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dummy Component renders correctly with required props 1`] = `
.c0 {
  color: red;
  width: 100%;
}

<div
  className="c0"
>
  My fake dummy prop is 
  40
</div>
`;
